### DISCUSSION TITLE
Current clone of project won't compile - include/effects/spectrumeffects.h:289:93: error: 'byte' has not been declared

### DISCUSSION THREAD

**Discussion by LarryXO** [PlummersSoftwareLLC/NightDriverStrip#132] [Q&A] [answered]
Can anyone point me in the direction of an answer?

I am on Win11, and am using what I think is the updated libraries for everything. I want to build the Spectrum with a M5StickPlus.

**Comment by damopinn** [comment, marked as answer]
I have the same problem, I saw the espressif framework was updated about a week ago, I'm trying to force version 3.5.0, I managed to compile about a month ago using that version.

**Reply by damopinn** [reply]
I changed every occurrence of  'platform = espressif32' to  'platform = espressif32@3.5.0' in platformio.ini.
This allows you to force the version that works.

**Reply by rbergen** [reply]
Just to make sure I understand correctly: this change does allow you to compile?

**Reply by damopinn** [reply]
I downloaded fresh and made this change and compiled successfully on windows 10 and on Manjaro.

**Reply by LarryXO** [reply]
Working on Windows 11 and Linux Mint...20, I think?

**Reply by damopinn** [reply]
[problems.txt](https://github.com/PlummersSoftwareLLC/NightDriverStrip/files/8588393/problems.txt)
It didn't work for me on Manjaro. This is the problems output. I couldnt work out how to get a logfile.

**Reply by rbergen** [reply]
To be honest, I've now lost track of what's (not) working for whom. 

@LarryXO did you comment on @damopinn's suggestion to pin down the espressif32 version, or my suggestion to add a `using`?
@damopinn did you compile with the espressif32 version still pinned to 3.5.0? Beyond that, I can't really "read" the problems output you shared, but I do see a lot of messages that seem entirely unrelated to the "undeclared 'byte'" issue.

**Reply by damopinn** [reply]
I did get it to work pinned to version 3.5.0. 
I did not get it work with the 'using' line. There were undeclared byte issues in lots of .h files.
I did not try to compile pinned and with the 'using' line.
Hope this clears up my part.

**Reply by rbergen** [reply]
@damopinn It does, thank you very much!

**Reply by LarryXO** [reply]
> To be honest, I've now lost track of what's (not) working for whom.
> 
> @LarryXO did you comment on @damopinn's suggestion to pin down the espressif32 version, or my suggestion to add a `using`? @damopinn did you compile with the espressif32 version still pinned to 3.5.0? Beyond that, I can't really "read" the problems output you shared, but I do see a lot of messages that seem entirely unrelated to the "undeclared 'byte'" issue.

Sorry, I'm new to this. I restricted the espressif32 to 3.5.0, and it worked. I also tried the namespace fix without restricting the espressif32 to 3.5.0, and although it did fix the spectrumeffects.h issue, there were many other issues with "byte" being referenced. 

I thought I had screwed something up in my path, or somewhere else, so I was asking for help with one of the last instances of the error. Thank you, @damopinn, and @rbergen.

**Reply by rbergen** [reply]
Ok, then I think it's indeed best to pin down espressif32 to version 3.5.0, at least until there is a pressing reason to upgrade the framework beyond that.
@damopinn could you put a PR together to apply this to the project?

**Reply by damopinn** [reply]
I created a PR with the changes, let me know if there's a problem it's my first time.

**Comment by rbergen** [comment]
**EDIT:** participant feedback shows that the following is not a fix for this issue.

I think the problem is that spectrumeffects.h lacks the following line after the `#include` directives:

```cpp
using namespace std;
```

It seems that a compatible `byte` declaration gets indirectly included with espressif32 platform versions up to and including 3.5.0, but not with higher versions. 

If anyone can confirm that adding the indicated `using` line does indeed fix the problem with the latest version of espressif32, then I will open a PR to update spectrumeffects.h accordingly.
